src/llvm-final-gc-lowering.cpp

```diff
@@ -277,6 +277,39 @@ bool FinalLowerGC::runOnFunction(Function &F)
     }
 
     return true;
+
+    verify_skip:
+#ifdef JL_VERIFY_PASSES
+    for (auto &BB : F) {
+        for (auto &I : make_early_inc_range(BB)) {
+            auto *CI = dyn_cast<CallInst>(&I);
+            if (!CI)
+                continue;
+
+            Value *callee = CI->getCalledOperand();
+            assert(callee);
+#define IS_INTRINSIC(INTRINSIC) \
+            do { \
+                auto intrinsic = getOrNull(jl_intrinsics::INTRINSIC); \
+                if (intrinsic == callee) { \
+                    errs() << "Final-GC-lowering didn't eliminate all intrinsics'" << F.getName() << "', dumping entire module!\n\n"; \
+                    errs() << *F.getParent() << "\n"; \
+                    abort(); \
+                } \
+            } while (0)
```

Comment:
an anonymous function would be better here:
```suggestion
            auto IS_INTRINSIC = [](auto intrinsic) {
                auto intrinsic = getOrNull(intrinsic);
                if (intrinsic == callee) {
                    errs() << "Final-GC-lowering didn't eliminate all intrinsics'" << F.getName() << "', dumping entire module!\n\n";
                    errs() << *F.getParent() << "\n";
                    abort();
                }
            };
```

Reply:
bump. should we land this, and rebase/retest after the llvm update?

Reply:
I think so, the nanosoldier changes seem to bit an overall improvement, but I'll rebase and retest